src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -151,6 +151,17 @@ public ParticipationResource(ParticipationService participationService, CourseSe
         }
 
         Participation participation = participationService.startExercise(exercise, principal.getName());
+
+        // Hide information from students
+        if (!courseService.userHasAtLeastTAPermissions(course)) {
```

Comment:
please use participation.getExercise().filterSensitiveInformation() instead. This is much better and handles all exercises properly